paasta_tools/cli/cmds/local_run.py

```diff
@@ -687,22 +687,18 @@ def configure_and_run_docker_container(
     )
 
 
-def local_makefile_present():
-    if makefile_responds_to('cook-image'):
-        sys.stderr.write("Local Makefile with 'cook-image' target detected. Assuming --build\n")
-        return True
-    else:
-        return False
-
-
 def paasta_local_run(args):
-    if (args.pull is None) and (args.build is None):
-        # Default to --build if neither is specified
-        if not local_makefile_present():
+    if args.pull is None and args.build is None:
```

Comment:
I think what we actually want:

```python
if args.dry_run:
    args.pull = args.build = False
elif ...:
    # Do the guesswork
```

It strikes me as odd that local-run does gueswork at all.  If I'm in an unrelated repository running `local-run -s yelp-main` and that repository *happens* to have a makefile suddenly the command is responding differently than otherwise.